src/packet/mod.rs

```diff
@@ -34,6 +34,12 @@ pub trait MutablePacket : Packet {
     }
 }
 
+/// Used to help compute checksum value of packet
+pub trait HasPseudoheader {
+    /// Converts IPv4 or IPv6 specific fields to a checksum value
+    fn pseudoheader_checksum(&self) -> u32;
```

Comment:
Since the trait name contains pseudo-header, there's no need to prefix this method - simply call it checksum.